src/main.rs

```diff
@@ -597,17 +601,28 @@ fn run() -> Result<()> {
             .context("Failed to build source distribution")?;
         }
         Opt::Pep517(subcommand) => pep517(subcommand)?,
-        Opt::Upload { files } => {
+        Opt::Upload { publish, files } => {
             if files.is_empty() {
                 println!("⚠  Warning: No files given, exiting.");
                 return Ok(());
             }
-            let metadata: Vec<Vec<(String, String)>> = files
+
+            // FIXME: All uploaded files are expected to share the build context
```

Comment:
I think the easiest way to check this is to read all metadata and check they are equal

Reply:
I've added a sanity check ensuring that all wheels come from the same Python package.